src/Symfony/Component/Console/Application.php

```diff
@@ -941,6 +942,19 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
             // ignore invalid options/arguments for now, to allow the event listeners to customize the InputDefinition
         }
 
+        if (extension_loaded('pcntl')) {
+            pcntl_async_signals(true);
+
+            $event = new ConsoleStopEvent($command, $input, $output);
+            $onStopHandler = function () use ($event) {
+                $this->dispatcher->dispatch($event, ConsoleEvents::STOP);
+                /** @TODO what exit code to return? */
+                exit;
```

Comment:
Instead of this, we can do:
Add a `ConsoleStopEvent::shouldStop()` merthod.

* If it returns false: do nothing
* If it returns true: trigger a `ConsoleTerminateEvent` and  `return $event->getExitCode();` (like below)